test/cypress/integration/fetchApi.spec.js

```diff
@@ -0,0 +1,82 @@
+describe('Fetch Api', () => {
+  it('Stubbed failed fetch API', () => {
+    cy.visit('/commands/network-requests');
+
+    cy.intercept(
+      {
+        method: 'PUT',
+        url: 'comments/*',
+      },
+      {
+        statusCode: 404,
+        body: {error: 'Test message.'},
+        delay: 500,
+      }
+    ).as('putComment');
+
+    cy.window().then((w) => {
+      fetch('/comments/10', {
+        method: 'PUT',
+        body: 'test',
+      });
+    });
+
+    cy.wait('@putComment');
+
+    cy.get('.breaking-get', {timeout: 1});
+  });
+
+  context('Real Fetch Requests', () => {
+    const testRealFetchRequest = (options) => {
+      cy.visit('/commands/network-requests');
+
+      if (options.interceptPath) {
+        // only intercepted fetch requests logs a response body. Read more at https://github.com/cypress-io/cypress/issues/17656
+        cy.intercept(options.interceptPath);
+      }
+
+      cy.window().then((window) => {
+        const document = window.document;
```

Comment:
What is the purpose of the button? Wouldn't it have been more easier to directly trigger `window.fetch(..)` here? And using and intercept await its response?

Reply:
window.fetch would result in a xhr query because of fetch-polyfill. This is also testing the case of not-intercepted requests.